examples/mnemonic_to_descriptors.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2020-2021 Bitcoin Dev Kit Developers
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+use bdk::bitcoin::secp256k1::Secp256k1;
+use bdk::bitcoin::util::bip32::{DerivationPath, KeySource};
+use bdk::bitcoin::Network;
+use bdk::keys::bip39::{Language, Mnemonic, WordCount};
+use bdk::keys::DescriptorKey::Secret;
+use bdk::keys::KeyError::Message;
+use bdk::keys::{DerivableKey, DescriptorKey, ExtendedKey, GeneratableKey, GeneratedKey};
+use bdk::miniscript::{miniscript, Segwitv0};
+use bdk::Error as BDK_Error;
+use std::error::Error;
+use std::str::FromStr;
+
+/// This example demonstrates how to generate a mnemonic phrase
+/// using BDK and use that to generate a master extended private key.
+/// It then uses the master extended private key with a derivation
+/// path to generate child extended private and public keys. Finally,
+/// it uses the derivation path and these keys to get the descriptors.
+fn main() -> Result<(), Box<dyn Error>> {
+    let secp = Secp256k1::new();
+
+    // In this example we are generating a 12 words mnemonic phrase
+    // but it is also possible generate 15, 18, 21 and 24 words
+    // using their respective `WordCount` variant.
+    let mnemonic: GeneratedKey<_, miniscript::BareCtx> =
```

Comment:
You should use the same ctx here for consistency (you used `Segwitv0` down below)

Reply:
Done